examples/run/run.cpp

```diff
@@ -1053,11 +1074,43 @@ static int get_user_input(std::string & user_input, const std::string & user) {
     return 0;
 }
 
+// Reads a chat template file to be used
+static std::string read_chat_template_file(const std::string & chat_template_file) {
+    if(chat_template_file.empty()){
+        return "";
+    }
+
+    FILE* file = ggml_fopen(chat_template_file.c_str(), "r");
+    if (!file) {
+        printe("Error opening chat template file '%s': %s", chat_template_file.c_str(), strerror(errno));
+        return "";
+    }
+
+    fseek(file, 0, SEEK_END);
+    size_t size = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    std::vector<unsigned char> data(size);
+    size_t                     read_size = fread(data.data(), 1, size, file);
+    fclose(file);
+    if (read_size != size) {
+        printe("Error reading chat template file '%s': %s", chat_template_file.c_str(), strerror(errno));
+        return "";
+    }
+    return std::string(data.begin(), data.end());
```

Comment:
One could actually fread directly to a std::string and save a copy. It's a optimisation, a small one. A std::string is basically a vector of chars.